library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -591,23 +591,33 @@ protected void replay(long windowId)
                 pendingFiles.remove(recoveryEntry.file);
               }
               inputStream = retryFailedFile(new FailedFile(recoveryEntry.file, recoveryEntry.startOffset));
+
```

Comment:
Actually, I take the first comment back: looks like the skipping in emitTuples might be needed for failed files (but not for replay) but please double check.